tests/Tokenizer/Transformer/TypeIntersectionTransformerTest.php

```diff
@@ -63,7 +63,7 @@ function foo(){}
             ',
         ];
 
-        if (\PHP_VERSION_ID >= 8_01_00) {
+        if (\defined('T_AMPERSAND_FOLLOWED_BY_VAR_OR_VARARG')) { // @TODO: drop condition when PHP 8.1+ is required
```

Comment:
for me, it is as unfortunate as checking PHP version directly , with very same reasoning. let's have dedicated provider.

Reply:
any idea how to do it right?

Reply:
Maybe the next iteration of refactoring could disallow `defined` in tests as well? Not much we have:
```
kub@:~/code/friendsofphp/PHP-CS-Fixer(master)$ git grep 'defined(' ./tests
tests/AutoReview/ProjectCodeTest.php:            if (\defined('T_ATTRIBUTE') && $tokens[$index]->isGivenKind(T_ATTRIBUTE)) {
tests/Fixer/Alias/NoAliasFunctionsFixerTest.php:            if (defined("'.$alias.'") || $'.$alias.' instanceof '.$alias.') {
tests/Fixer/Alias/RandomApiMigrationFixerTest.php:        if (!defined("srand") || $srand instanceof srand) {
tests/Fixer/Basic/PsrAutoloadingFixerTest.php:            if (\defined($tokenType)) {
tests/Tokenizer/Analyzer/ControlCaseStructuresAnalyzerTest.php:        if (\defined('T_ENUM')) { // @TODO: drop condition when PHP 8.1+ is required - sadly PHPUnit still calls the provider even if requires condition is not matched
tests/Tokenizer/Analyzer/ControlCaseStructuresAnalyzerTest.php:        if (\defined('T_MATCH')) { // @TODO: drop condition when PHP 8.0+ is required - sadly PHPUnit still calls the provider even if requires condition is not matched
tests/Tokenizer/CTTest.php:        self::assertFalse(\defined($name), 'The CT name must not use native T_* name.');
tests/Tokenizer/TokenTest.php:        if (\defined('T_ATTRIBUTE')) {
tests/Tokenizer/TokenTest.php:        if (\defined('T_NULLSAFE_OBJECT_OPERATOR')) {
tests/Tokenizer/TokenTest.php:        if (\defined('T_AMPERSAND_FOLLOWED_BY_VAR_OR_VARARG')) { // @TODO: drop condition when PHP 8.1+ is required
tests/Tokenizer/TokenTest.php:        if (\defined('T_ENUM')) {
tests/Tokenizer/Transformer/NameQualifiedTransformerTest.php:        if (\defined('T_NAME_RELATIVE')) { // @TODO: drop condition when PHP 8.0+ is required
```
and most of them share the same problem - using undefined constants (like `T_ENUM` in PHP 7.4) even in data provider fails the PHPUnit.

Reply:
most/all of them can be replaced with checking PHP version.
OK to have it in another iteration.